### DISCUSSION TITLE
Any way to add Margin/Padding/Border to items in SelectableTextBlock?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#13472] [Q&A]
Hi,

I'm experimenting with `SelectableTextBlock` and have discovered I can add text using the `Inlines` property, which is fab. This means I can set fonts and colors by adding elements of the `Run` class, as below.

However, I really need to add items with margins and padding (and hopefully border). I realise I can add a `Control` also, but when I do this the copy text (CTRL-C) does not work.

Is there any way to set margins on a Run item? Or some other way?

Cheers

        // Texty is type of SelectableTextBlock

        var t = new Run("Hello World\n");
        t.FontSize = 32;
        Texty.Inlines.Add(t);

        t = new Run("Hello World2");
        t.FontSize = 20;
        Texty.Inlines.Add(t);

**Comment by maxkatz6** [comment]
You can use InlineUIContainer to embed complex controls inside

**Comment by kuiperzone** [comment]
Thank you for the reply.

The problem with that is the select does not work on contained SelectableTextBlocks, and the parent ceases to be a SelectableTextBlock but merely a TextBlock. Moreover, the select operation on any simple inline text portions becomes buggy when there are contained controls in the content also (am planning on raising a bug at some point).

If select worked fully for contained SelectableTextBlock controls, then there's a use case for this -- it could be used as the basis of a HTML renderer (users would expect to be able to select and copy text so a vanilla TextBlock won't cut it).

I'm fully tied up at the moment, but am planning to come back with some input.
